aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,362 @@
+							ForceNew:     true,
+						},
+						"instance_types": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+							ForceNew: true,
+						},
+						"max_vcpus": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"min_vcpus": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"security_group_ids": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+							ForceNew: true,
+						},
+						"spot_iam_fleet_role": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"subnets": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+							ForceNew: true,
+						},
+						"tags": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							ForceNew: true,
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringInSlice([]string{batch.CRTypeEc2, batch.CRTypeSpot}, true),
+						},
+					},
+				},
+			},
+
+			"service_role_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+
+			"state": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{batch.CEStateEnabled, batch.CEStateDisabled}, true),
+			},
+
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{batch.CETypeManaged, batch.CETypeUnmanaged}, true),
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsBatchComputeEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+	input := batch.CreateComputeEnvironmentInput{
+		ComputeEnvironmentName: aws.String(d.Get("name").(string)),
+		ServiceRole:            aws.String(d.Get("service_role_arn").(string)),
+		State:                  aws.String(d.Get("state").(string)),
+		Type:                   aws.String(d.Get("type").(string)),
+	}
+	input.ComputeResources = createComputeResource(d)
+	out, err := conn.CreateComputeEnvironment(&input)
+	name := d.Get("name").(string)
+	if err != nil {
+		return fmt.Errorf("%s %q", err, name)
+	}
+
+	log.Println(
+		"[INFO] Waiting for ComputeEnvironment to be available")
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{batch.CEStatusCreating, batch.CEStatusUpdating},
+		Target:     []string{batch.CEStatusValid},
+		Refresh:    batchComputeEnvironmentRefreshFunc(conn, name),
+		Timeout:    10 * time.Minute,
```

Comment:
They're hardcoded everywhere else in the repo so I am going with what is standard practice.